lib/palindrome_check.rb

```diff
@@ -1,5 +1,72 @@
 # A method to check if the input string is a palindrome.
 # Return true if the string is a palindrome. Return false otherwise.
+
+###all spaces must be ignored!!!
+
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  # raise NotImplementedError
+  return false if my_phrase == nil
+  return true if my_phrase.length == 1
+
+  i = 0
+  j = my_phrase.length - 1
+
+  # puts palindrome_check("han nah  ")
+
+  while i < j do
+    if my_phrase[i] == my_phrase[j]
```

Comment:
I'd recommend taking care of preceding and trailing white spaces before comparing the two characters. That way when you do compare the character at index _i_ with the character at index _j_, you can be sure that they are not equal.